src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -786,18 +786,29 @@ pSvg = do
 pCodeWithClass :: PandocMonad m => Text -> Text -> TagParser m Inlines
 pCodeWithClass name class' = try $ do
   TagOpen open attr' <- pSatisfy $ tagOpen (== name) (const True)
-  result <- manyTill pAny (pCloses open)
   let (ids,cs,kvs) = toAttr attr'
       cs'          = class' : cs
-  return . B.codeWith (ids,cs',kvs) .
-    T.unwords . T.lines . innerText $ result
+  code open (ids,cs',kvs)
 
 pCode :: PandocMonad m => TagParser m Inlines
 pCode = try $ do
   (TagOpen open attr') <- pSatisfy $ tagOpen (`elem` ["code","tt"]) (const True)
   let attr = toAttr attr'
-  result <- manyTill pAny (pCloses open)
-  return $ B.codeWith attr $ T.unwords $ T.lines $ innerText result
+  code open attr
+
+code :: PandocMonad m => Text -> Attr -> TagParser m Inlines
+code open attr = do
+  exts <- getOption readerExtensions
+  -- Don't use `formatCode` in an epub because it can incorrectly parse certain
+  -- elements. For example, `<code>epub:switch</code>` is parsed as a link to
+  -- the switch section of the epub spec.
```

Comment:
I'm confused about this comment. Does "it" in the first line refer to `formatCode` or EPUB?  In either case, can you explain further?